go_modules

```diff
@@ -227,7 +227,14 @@ def main():
 
     archive_args = get_archive_parameters(args)
     vendor_tarname = f"{archive_args['vendorname']}.{archive_args['ext']}"
-    archive = args.archive or archive_autodetect()
+    if args.archive:
+        archive_matches = sorted(Path.cwd().glob(args.archive), reverse=True)
```

Comment:
I'm not sure why this service uses this `reverse=True` and silently only using the first entry pattern for globbing, but I've replicated it here for consistency.

Reply:
Thanks for the PR, will merge. 

The pattern of `reverse=True` and selecting the first entry is there to select the one highest version of similarly named archives that can be temporarily present in the working directory when using tar_scm . Lexicographic sorting is incorrect in some cases e.g. `v0.10.0 < v0.2.0`. Without adding on a python library dependency, my goal is to find a semantic version comparison function(s) that works acceptably with Go version strings and archive filename versions including git refs/offsets, then replace both occurrences of this pattern.